### PR TITLE
Retiring endpoints referring to member by stable_id only

### PR DESCRIPTION
**Pull request by thiagogenez**
### Description

At the end of their deprecation period, Compara REST endpoints referring to member by stable_id will need to be removed.

https://www.ebi.ac.uk/panda/jira/browse/ENSCOMPARASW-6261

### Use case

The following Compara REST endpoints referring to member by stable_id have been removed:

- Endpoint "GET cafe/genetree/member/id/:id"
- Endpoint "GET genetree/member/id/:id"
- Endpoint "GET homology/id/:id"

### Benefits

The retirement of deprecated REST functions

### Possible Drawbacks

None

### Testing

note to submitters and reviewers: documentation-only changes may reflect
changes in other repos that can result in new or different output from
REST endpoints. In turn, these may require new tests or changes to existing tests.

_Have you added/modified unit tests to test the changes?_

_If so, do the tests pass/fail?_
Pass

_Have you run the entire test suite and no regression was detected?_
I haven't.

### CODE REVIEW

File: t/homology.t [hunk 1]
```diff
@@ -59,57 +59,51 @@ my $condensed_ortho_ENSG00000139618_gorilla = {
...
 is(scalar(@{$json->{data}->[0]->{homologies}}), 2, 'We get all the one-to-many homologies even though there is 1 species');
 
 
 ## Total number of homologies without a filter
 $json = json_GET(
-    '/homology/id/ENSG00000139618?compara=homology;format=condensed;type=orthologues',
+    '/homology/id/homo_sapiens/ENSG00000139618?compara=homology;format=condensed;type=orthologues',
     '"condensed" homologies with no species filter',
 );
 is(scalar(@{$json->{data}->[0]->{homologies}}), 63, 'Got all the homologies');
```
Comment (twalsh-ebi) [inline t/homology.t:109]: I think this unit test may be the same as the one immediately following it. If the two unit tests are identical, would it be possible to delete one of them?

File: t/gene_tree.t [hunk 2]
```diff
@@ -338,22 +326,6 @@ sub find_leaf {
     }
 }
 
-my $json1 = json_GET(
-    '/genetree/member/id/ENSG00000176515?compara=homology&clusterset_id=default',
-    'Gene-tree (ncRNA) by ID. Explicitly require the default clusterset_id',
-);
-is(count_leaves($json1->{tree}), 69, 'Got all the leaves');
-my ($human_leaf) = find_leaf($json1->{'tree'}, 'ENSG00000176515');
```
Comment (twalsh-ebi) [inline t/gene_tree.t:346]: If `my $human_leaf` is used by any of the remaining unit tests, it may need to be declared where it is first used.

**Review by twalsh-ebi — COMMENTED** [review]
Thanks for this @thiagogenez. I just have a couple of questions/comments.

**Comment by thiagogenez** [comment]
Test done 

```bash
 prove gene_tree.t
gene_tree.t .. Warning: Using a password on the command line interface can be insecure.
gene_tree.t .. ok
All tests successful.
Files=1, Tests=51, 22 wallclock secs ( 0.02 usr  0.01 sys +  2.38 cusr  0.47 csys =  2.88 CPU)
Result: PASS
```

**Comment by thiagogenez** [comment]
Thanks @twalsh-ebi  - fixed declaration issue and removed the duplicated test

**Review by twalsh-ebi — APPROVED** [review]
Thanks for responding to my feedback @thiagogenez. Happy to approve.

**Review by nwillhoft — APPROVED** [review]
PR relates to the retirement of deprecated REST functions. Approving based on checks passing and code review carried out by Compara team.

**Pull request merged**